app/models/spree/order_decorator.rb

```diff
@@ -367,6 +371,7 @@ def ensure_customer
   end
 
   def update_adjustment!(adjustment)
+    return if adjustment.finalized?
```

Comment:
This actually makes sense for shipments as well. If something has been shipped, the shipping costs occurred and should not be changed.

The only case that may complicate things:
- Alice orders a, b and c from a shop and pays shipping fees depending on the number of items.
- The shop sends a box to Alice.
- Alice discovers that item c is missing and complains.
- The shop takes item c out of the original order, shipping fees are not updated.
- The shop has to manually add a credit or refund.
- The previous code would have adjusted the fee and the customer's balance would change.

Reply:
Good point. Do you think we need to preserve this particular case, or is it rare enough that we don't need to support it?

The fact that this case would "work" as it currently stands is an artefact of these `after_save` callbacks which we added, right? I'm not sure that handling this case was part of the design. As far as I know, doing the same thing in an unaltered deployment of Spree would not cause the shipping fee to be updated.

Reply:
I think it's rare enough. I don't think it was the initial intent of this code, just a by-product. So I'm happy for this change to got through.

Reply:
Just tested this. This case isn't even possible, because removing shipped items from an order is not allowed:

![screen shot 2017-10-20 at 12 49 26 pm](https://user-images.githubusercontent.com/2067859/31801647-3fba84d4-b595-11e7-8cb3-6b16d96a28fd.png)